pkg/scheduler/eventhandlers.go

```diff
@@ -122,11 +124,11 @@ func (sched *Scheduler) updateNodeInCache(oldObj, newObj interface{}) {
 
 	// Only activate unschedulable pods if the node became more schedulable.
 	// We skip the node property comparison when there is no unschedulable pods in the queue
-	// to save processing cycles. We still trigger a move to active queue to cover the case
-	// that a pod being processed by the scheduler is determined unschedulable. We want this
-	// pod to be reevaluated when a change in the cluster happens.
-	if sched.SchedulingQueue.NumUnschedulablePods() == 0 || nodeSchedulingPropertiesChanged(newNode, oldNode) {
-		sched.SchedulingQueue.MoveAllToActiveQueue()
+	// to save processing cycles.
+	if sched.SchedulingQueue.NumUnschedulablePods() != 0 {
```

Comment:
Please pay attention to this change.
The previous comment/code doesn't make sense to me. If `NumUnschedulablePods` is 0, when you call `MoveAllToActiveQueue`, it scans the `unschedulableQ` map, which should be empty.